Cmake fails to find readline library

## Context
Trying to run cmake to build neuron and it fails to find libreadline

### Overview of the issue
Trying to build neuron from scratch so I can have CoreNeuron support, GPU processing on Ubuntu 20 system
The initial cmake command fails (command given below at bottom) saying it can't find readline.  I apt-get installed libreadline-dev and ldconfig -p |more outputs:

libreadline.so.8 (libc6,x86-64) => /lib/x86_64-linux-gnu/libreadline.so.8
libreadline.so.5 (libc6,x86-64) => /lib/x86_64-linux-gnu/libreadline.so.5

## Logs


cmake .. -DNRN_ENABLE_CORENEURON=ON -DCORENRN_ENABLE_GPU=ON -DNRN_ENABLE_INTERVIEWS=OFF -DNRN_ENABLE_RX3D=OFF -DCMAKE_INSTALL_PREFIX=/opt/nrn -DCMAKE_C_COMPILER=nvc -DCMAKE_CXX_COMPILER=nvc++ -DCMAKE_CUDA_COMPILER=nvcc
-- The C compiler identification is PGI 22.1.0
-- The CXX compiler identification is PGI 22.1.0
-- Check for working C compiler: /opt/nvidia/hpc_sdk/Linux_x86_64/22.1/compilers/bin/nvc
-- Check for working C compiler: /opt/nvidia/hpc_sdk/Linux_x86_64/22.1/compilers/bin/nvc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/nvidia/hpc_sdk/Linux_x86_64/22.1/compilers/bin/nvc++
-- Check for working CXX compiler: /opt/nvidia/hpc_sdk/Linux_x86_64/22.1/compilers/bin/nvc++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
heads/master
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Found BISON: /usr/bin/bison (found version "3.5.1") 
-- Found FLEX: /usr/bin/flex (found version "2.6.4") 
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find Readline (missing: Readline_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindReadline.cmake:45 (find_package_handle_standard_args)
  CMakeLists.txt:189 (find_package)


-- Configuring incomplete, errors occurred!

--- a/cmake/modules/FindReadline.cmake
+++ b/cmake/modules/FindReadline.cmake
@@ -27,10 +27,18 @@ find_path(
   NAMES readline/readline.h
   HINTS ${Readline_ROOT_DIR}/include)
 
+# On platforms like debian/ubuntu with multi-arch support
+# libraries might be under separate directory like /usr/lib/x86_64-linux-gnu/
+# With NVHPC compilers CMake can not find these librarise, see
+# https://forums.developer.nvidia.com/t/nvhpc-cmake-issues-cmake-library-architecture-is-not-set-and-hence-system-libraries-are-not-found/200659
+# Hence, provide extra directories as hint to search for libraries
+file(GLOB EXTRA_SEARCH_DIRS
+  "${Readline_ROOT_DIR}/lib*/${CMAKE_SYSTEM_PROCESSOR}-*-*")
+
 find_library(
   Readline_LIBRARY
   NAMES readline
-  HINTS ${Readline_ROOT_DIR}/lib)
+  HINTS ${Readline_ROOT_DIR}/lib ${EXTRA_SEARCH_DIRS})
 
 if(Readline_INCLUDE_DIR
    AND Readline_LIBRARY

--- a/docs/cmake_doc/options.rst
+++ b/docs/cmake_doc/options.rst
@@ -372,10 +372,17 @@ NRN_NMODL_CXX_FLAGS:STRING=""
 
 Readline_ROOT_DIR:PATH=/usr
 ---------------------------
-  Path to a file.  
+  Install directory prefix where readline is installed.
 
-  If cmake can't find readline, you
-can give this hint as to where it is.
+  If cmake can't find readline, you can give this hint with the directory
+  path under which readline is installed. Note that on some platforms
+  with multi-arch support (e.g. Debian/Ubuntu), CMake versions < 3.20 are not
+  able to find readline library when NVHPC/PGI compiler is used (for GPU
+  support). In this case you can install newer CMake (>= 3.20) or explicitly
+  specify readline library using `-DReadline_LIBRARY=` option:
+  .. code-block::
+
+    -DReadline_LIBRARY=/usr/lib/x86_64-linux-gnu/libreadline.so
 
 NRN_ENABLE_TESTS:BOOL=OFF
 -------------------------

